src/applications/find-forms/containers/SearchResults.jsx

```diff
@@ -23,62 +23,78 @@ import { FAF_SORT_OPTIONS, FAF_TEST_OPTION_CLOSEST_MATCH } from '../constants';
 import SearchResult from '../components/SearchResult';
 
 export const MAX_PAGE_LIST_LENGTH = 10;
-export class SearchResults extends Component {
-  static propTypes = {
-    // From mapStateToProps.
-    error: PropTypes.string.isRequired,
-    fetching: PropTypes.bool.isRequired,
-    page: PropTypes.number.isRequired,
-    query: PropTypes.string.isRequired,
-    results: PropTypes.arrayOf(customPropTypes.Form.isRequired),
-    hasOnlyRetiredForms: PropTypes.bool.isRequired,
-    sortByPropertyName: PropTypes.string,
-    startIndex: PropTypes.number.isRequired,
-    useLighthouseSearchAlgo: PropTypes.bool,
-    // From mapDispatchToProps.
-    updateSortByPropertyName: PropTypes.func,
-    updatePagination: PropTypes.func.isRequired,
-  };
+const usePreviousProps = value => {
+  // This is a mirror to storing and assessing prevProps vs current props
+  const ref = useRef(); // Refs are like a class instance var, in react their values don't change unless that same ref is redefined.
+  useEffect(() => {
+    // This hook is run ONLY after the return statement of the component is evaluated (ComponentDidMount || Update)
+    ref.current = value;
+  });
+  return ref.current; // Returns the value BEFORE the return statement of the component is evaluated.
+};
+/** How it works ^
+ * Essentially it's step by step.
+ * 1. Init of SearchResults which calls usePreviousProps() defining the REF. (Ignores UseEffect because component is not rendered yet)
+ * 2. SearchResults return statement is evaluated.
+ * 3. (CDM) UseEffects is evoked and the variables are assessed. PrevProps is still the same as when it was evoked during step 1.
+ * 4. Consumer fills in the search bar and hits search. That updates Redux which returns NEW Props to component.
+ * 5. (CDU) usePreviousProps(newProps) is invoked; it returns the value of the REF from step 1 is return as it is the "Previous Props".
+ * 6. #2 is repeated.
+ * 7. (CDU) UseEffects is evoked and the variables are assessed. PrevProps is still the same as when it was evoked during step 1. However, the "newProps" from step #5 will now be the new "prevProps" and the REF Value is overwritten.
+ * 8. Repeat
+ */
+
+export const SearchResults = ({
+  error,
+  fetching,
+  page,
+  query,
+  results,
+  sortByPropertyName,
+  hasOnlyRetiredForms,
+  startIndex,
+  useLighthouseSearchAlgo,
+  updatePagination,
+  updateSortByPropertyName,
+}) => {
+  const prevProps = usePreviousProps({
```

Comment:
So the object we're passing here is a current ref? I'm a little confused by this but it's nice that we're moving to hooks!

Reply:
Yeah, it's really confusing but it's the only way I found where we can Mirror the behavior. 
@kelsonic here's what I found in the React docs. 
https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state